swaggen/src/main/java/generator/PathGenerator.java

```diff
@@ -54,9 +53,49 @@
 	 * @param annotation the annotation
 	 * @return the path
 	 */
-	private static Map<RequestMethod, Endpoint> generatePathFromAnnotation(SwaggerGen annotation) {
-		Map<RequestMethod, Endpoint> endpointMap = new HashMap<>();
-		endpointMap.put(RequestMethod.valueOf(annotation.method()), EndpointFactory.createEndpoint(annotation));
-		return endpointMap;
+	private static SwaggerEndpoint generatePathFromAnnotation(SwaggerGen annotation) {
+		SwaggerEndpoint endpoint = new SwaggerEndpoint();
+		endpoint.addEndpoint(RequestMethod.valueOf(annotation.method()), EndpointFactory.createEndpoint(annotation));
+		return endpoint;
+	}
+
+	/**
+	 * Checks if URL is already in the existing map, and if it is, appends the swagger
+	 * endpoints to the URL
+	 * Case: multiple classes with the same endpoint
+	 * @param existingPaths Map of existing Swagger Endpoints
+	 * @param classPaths Map of new Swagger Endpoints in the class
+	 */
+	private static void checkClassEndpoints(Map<String, SwaggerEndpoint> existingPaths, Map<String, SwaggerEndpoint> classPaths) {
+		if (!existingPaths.isEmpty()) {
+			for(Map.Entry<String, SwaggerEndpoint> pathEntry: existingPaths.entrySet()) {
+				for(Map.Entry<String, SwaggerEndpoint> classEntry: classPaths.entrySet()) {
+					if (pathEntry.getKey().equals(classEntry.getKey())) {
+						pathEntry.getValue().addEndpoint(classEntry.getValue());
+					} else {
+						existingPaths.putAll(classPaths);

```

Comment:
Why are we adding all of the classPaths to the existingPaths for every entry in every class path and for each existing path? This seems superfluous at best, but it looks more like a bug. 

Add a unit test (if there is not one already) for that calls this method with multiple class paths. At least 2 should have keys that match paths in the existingPaths. Then make sure the behaviour is correct.

Reply:
This block of code allows for multiple classes with the same url but different request methods to be appended to the map properly. Otherwise, they would just overwrite each other in the map.

There is already a unit test that tests three classes (MockMinimalGetEndpoint, MockMinimalPostEndpoint, MockMinimalPutEndpoint) that have the same url, and the behaviour is correct because it was able to be appended properly.

Reply:
This appears to have been fixed when the maps were flattened. The issue wasn't that `existingPaths.putAll(classPaths);` was being called, but rather that it was being called once for every entry for classPaths rather than just once.